stubs/setuptools/setuptools/command/test.pyi

```diff
@@ -1,43 +1,16 @@
-from _typeshed import Incomplete, Unused
-from collections.abc import Callable
-from types import ModuleType
-from typing import ClassVar, Generic, TypeVar, overload
-from typing_extensions import Self
-from unittest import TestLoader, TestSuite
+from typing import ClassVar
+from typing_extensions import deprecated
 
 from .. import Command
 
-_T = TypeVar("_T")
-_R = TypeVar("_R")
-
-class ScanningLoader(TestLoader):
-    def __init__(self) -> None: ...
-    def loadTestsFromModule(self, module: ModuleType, pattern: Incomplete | None = None) -> list[TestSuite]: ...  # type: ignore[override]
-
-class NonDataProperty(Generic[_T, _R]):
-    fget: Callable[[_T], _R]
-    def __init__(self, fget: Callable[[_T], _R]) -> None: ...
-    @overload
-    def __get__(self, obj: None, objtype: Unused = None) -> Self: ...
-    @overload
-    def __get__(self, obj: _T, objtype: Unused = None) -> _R: ...
-
-class test(Command):
-    description: str
+@deprecated(
+    "The test command is disabled and references to it are deprecated. Please remove any references to `setuptools.command.test` in all supported versions of the affected package."
+)
+class _test(Command):
+    description: ClassVar[str]
     user_options: ClassVar[list[tuple[str, str, str]]]
-    test_suite: Incomplete
-    test_module: Incomplete
-    test_loader: Incomplete
-    test_runner: Incomplete
     def initialize_options(self) -> None: ...
     def finalize_options(self) -> None: ...
-    @NonDataProperty
-    def test_args(self) -> list[str]: ...
-    def with_project_on_sys_path(self, func) -> None: ...
-    def project_on_sys_path(self, include_dists=()): ...
-    @staticmethod
-    def paths_on_pythonpath(paths) -> None: ...
-    @staticmethod
-    def install_dists(dist): ...
     def run(self) -> None: ...
-    def run_tests(self) -> None: ...
+
+test = _test
```

Comment:
This doesn't seem to propagate the static deprecation (at least not in pylance). Given it's a private name anyway, I'd just mark `test` directly.
![image](https://github.com/user-attachments/assets/5a43c23d-787e-4adb-a310-b2f59671afc7)

Reply:
The problem with that is that we'd have to ignore the stubtest errors and stubtest won't complain if the class is removed or we make an error.

Reply:
Are you sure? I'm running stubtest on setuptools locally on your branch:
```py
@deprecated(
    """\
The test command is disabled and references to it are deprecated. \
Please remove any references to `setuptools.command.test` in all supported versions of the affected package.\
"""
)
class test(Command):
    description: ClassVar[str]
    user_options: ClassVar[list[tuple[str, str, str]]]
    def initialize_options(self) -> None: ...
    def finalize_options(self) -> None: ...
    def run(self) -> None: ...
```

And it's passing as-is

Reply:
@srittau Setuptools 73 has just been released. I'd like to sync the typeshed stubs to it as well.

If you still feel unsure about my suggestion, I can approve your PR as-is and make the suggestion separatly.